students/DanCornutt/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,138 @@
+""""
+HW5 HP Norton MongoDB Project
+
+"""
+import csv
+import logging
+from pymongo import MongoClient
+
+
+class MongoDBConnection():
+    """MongoDB Connection"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        """ be sure to use the ip address not name for local windows"""
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def log_setup():
+    log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s \
+    %(message)s"
+    logging.basicConfig(level=logging.WARNING, format=log_format,
+                        filename='mylog.log')
+
+
+def show_available_products():
+    """Returns dict of products available for rental"""
+    info = {}
+    qset = db['products'].find({"quantity_available": {"$gt": "0"}})
+    for item in qset:
+        id = item['product_id']
+        content = dict((k, item[k]) for k in (
+            'description', 'product_type', 'quantity_available'
+        ))
```

Comment:
Nice. Did you know dict can take longer than doing a comprehension?
```
content = {k: item[k] for k in ('description', 'product_type', 'quantity_available')}
```
Also, some might find ```info[id] = content``` more clear than info.update.